### ISSUE TITLE
QGIS-3 install: couldn't load SIP module -- PqQt5 problem?

### ISSUE THREAD

**Issue by sirosenberg** [OSGeo/homebrew-osgeo4mac#510]
I just installed QGIS3 (3.4.0) using `brew install qgis3` on Mac (High Sierra 10.13.6).   No errors, but a warning: `osgeo/osgeo4mac/qgis3 dependency gcc was built with a different C++ standard library (libstdc++ from clang). This may cause problems at runtime.`  

When I open /usr/local/opt/qgis3/QGIS.app, I get the error: `Couldn't load SIP module.
Python support will be disabled.`  The error message says `ModuleNotFoundError: No module named 'PyQt5.QtCore'`.  [full text below]

Based on some other issues, I checked and got the following: 
`Already linked: /usr/local/Cellar/pyqt/5.10.1_1`
`osgeo/osgeo4mac/pyqt5-webkit 5.10.1 is already installed and up-to-date` 
`osgeo/osgeo4mac/qt5-webkit 5.11.2 is already installed and up-to-date`

Not sure if relevant, but I had to `brew unlink pyqt-qt4` before installing qgis3.

Ideas?

```
Traceback (most recent call last):
  File "", line 1, in 
  File "/usr/local/Cellar/qgis3/3.4.0/QGIS.app/Contents/MacOS/../Resources/python/qgis/__init__.py", line 72, in 
    from qgis.PyQt import QtCore
  File "/usr/local/Cellar/qgis3/3.4.0/QGIS.app/Contents/MacOS/../Resources/python/qgis/PyQt/QtCore.py", line 26, in 
    from PyQt5.QtCore import *
ModuleNotFoundError: No module named 'PyQt5.QtCore'


Python version:
3.7.0 (default, Oct  2 2018, 09:20:07) 
[Clang 10.0.0 (clang-1000.11.45.2)]

QGIS version:
3.4.0-Madeira 'Madeira', exported

Python path:
['/usr/local/Cellar/qgis3/3.4.0/QGIS.app/Contents/MacOS/../Resources/python', '~/Library/Application Support/QGIS/QGIS3/profiles/default/python', '~/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins', '/usr/local/Cellar/qgis3/3.4.0/QGIS.app/Contents/MacOS/../Resources/python/plugins', '/usr/local/opt/gdal2-python/lib/python3.7/site-packages', '/usr/local/opt/qgis3/lib/python3.7/site-packages', '/usr/local/opt/qgis3/libexec/vendor/lib/python3.7/site-packages', '/usr/local/Cellar/qgis3/3.4.0/libexec/vendor/lib/python3.7/site-packages', '/Users/steven', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python37.zip', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload']```

**Comment by fjperini** [comment]
I think the problem is in `sip-qt4` and `pyqt-qt4`. If you do not use it then try unlinked.

@nickrobison we would need to add the module in `sip`: `"--sip-module=PyQt5.sip"` [[see more information](https://github.com/OSGeo/homebrew-osgeo4mac/issues/481#issuecomment-431729132)]

In the new versions of pyqt (5.11.3) and `sip` (4.19.13) that are not yet in `homebrew/core`, several problems arise. It is necessary to inform this problem to the developers @nyalldawson so that we do not have problems with these modules.

**Comment by fjperini** [comment]
@nickrobison Yes I agree. I already have the new formulas of `sip`, `pyqt` and `pyqt5-webkit`. I will apply the same changes to `sip-qt4` and `pyqt-qt4`.

**Comment by fghorow** [comment]
Any workarounds for this? This is still biting me as of right now...

**Comment by fjperini** [comment]
@fghorow I will check this!

**Comment by Wicky2k8** [comment]
Any word?

**Comment by fjperini** [comment]
@sirosenberg @fghorow @Wicky2k8 You can try the new formula we are testing, with this the problem you are having should be solved.  See #607

**Comment by luispuerto** [comment]
I fixed a similar problem (see https://github.com/OSGeo/homebrew-osgeo4mac/issues/607#issuecomment-460920980) with:

```shell
$ brew reinstall ninja gsl sip six bison flex pkg-config python qt pyqt qscintilla2
$ brew link --overwrite pyqt
```
Worth a try.

**Comment by fjperini** [comment]
@sirosenberg @fghorow @Wicky2k8 @joh10891 Now you can try the new revision of the formula.

**Comment by fjperini** [comment]
Some useful suggestions.

In case you have installed another version of Python (Anaconda, Miniconda or if you used the installer provided by python.org).
QGIS will use the first Python that is in its PATH, so the installation may fail.

If the installation failed due to the problem reported in https://github.com/OSGeo/homebrew-osgeo4mac/issues/520

Try after doing:

       $ brew unlink python && brew link --force python

       $ $PATH (Check that there is no other version)

If the installation failed due to the problem reported in [#607 (comment)](https://github.com/OSGeo/homebrew-osgeo4mac/issues/607#issuecomment-460920980)

Try after doing:

       $ brew reinstall ninja gsl sip six bison flex pkg-config python qt pyqt qscintilla2

       $ brew link --overwrite pyqt

Other Notes:

An installation that failed previously may have created this link

       /usr/local/lib/python3.7/site-packages/PyQt5/uic/widget-plugins/qgis_customwidgets.py

you will need to delete it.

Also make sure that the folder `/usr/local/Cellar/qgis` does not exist if the installation fails.

It is also recommended remove the Homebrew Cache `$ rm -rf $(brew --cache)` (`~/Library/Caches/Homebrew`) and the temporary build files in `/tmp`.

**Comment by fjperini** [comment]
@sirosenberg @fghorow @Wicky2k8 @joh10891 formula. With all the restrictions there should no longer be any problem with these modules.
If it fails for the same reason, it is because you need to check your Homebrew. if it is compiled in Travis CI, it should be built on any machine.

A bottle will be built, if you only install qgis it without extra supports, you will not need to build.

**Comment by eldang** [comment]
I'm running into this problem again.  I tried all the workarounds suggested above and it's still happening.  From other things I'm reading it may be because homebrew has updated Python to 3.7.x; will investigate that further tomorrow.

**Comment by dd-a** [comment]
I have the same problem.  I did a brew upgrade yesterday and since then I have this message 

`Traceback (most recent call last):
  File "", line 1, in 
  File "/usr/local/opt/osgeo-qgis-ltr/QGIS.app/Contents/MacOS/../Resources/python/qgis/__init__.py", line 72, in 
    from qgis.PyQt import QtCore
  File "/usr/local/opt/osgeo-qgis-ltr/QGIS.app/Contents/MacOS/../Resources/python/qgis/PyQt/QtCore.py", line 26, in 
    from PyQt5.QtCore import *
ModuleNotFoundError: No module named 'PyQt5.sip'


Version de Python :
3.7.3 (default, Mar 27 2019, 09:23:39) 
[Clang 10.0.0 (clang-1000.11.45.5)]

Version de QGIS :
3.4.7-Madeira 'Madeira', exported

Chemin vers Python :
['/usr/local/opt/osgeo-qgis-ltr/QGIS.app/Contents/MacOS/../Resources/python', '/Users/didier/Library/Application Support/QGIS/QGIS3/profiles/default/python', '/Users/didier/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins', '/usr/local/opt/osgeo-qgis-ltr/QGIS.app/Contents/MacOS/../Resources/python/plugins', '/usr/local/opt/osgeo-gdal-python/lib/python3.7/site-packages', '/usr/local/opt/osgeo-qgis-ltr/QGIS.app/Contents/Resources/python', '/usr/local/opt/osgeo-qgis-ltr/lib/python3.7/site-packages', '/usr/local/opt/lib/python3.7/site-packages', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python37.zip', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload', '/Users/didier/Library/Python/3.7/lib/python/site-packages', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages', '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/geos', '/usr/local/Cellar/osgeo-matplotlib/3.0.3_1/libexec/lib/python3.7/site-packages', '/usr/local/Cellar/numpy/1.16.3_1/libexec/nose/lib/python3.7/site-packages', '/usr/local/Cellar/protobuf/3.7.1/libexec/lib/python3.7/site-packages']`

Qgis-ltr was running fine before that and QGIS 3.6.2 is not affected.
The upgrade did 
`swig 3.0.12 -> 4.0.0, qt 5.12.2 -> 5.12.3, osgeo/osgeo4mac/osgeo-qgis-ltr 3.4.6_1 -> 3.4.7_3, osgeo/osgeo4mac/osgeo-postgresql 11.2_1 -> 11.2_2, poppler 0.74.0 -> 0.76.0`

**Comment by BorisDess** [comment]
Hello. 
I got this problem once, tried to solve it with fixes online, at the end I couldn't even install python, I had fucked my system.
I reinstalled a brand new system.
Installed Anaconda and QGIS.
Worked for a couple of days, but then, probably because I installed Qt Creator, it stopped working.
I reinstalled QGIS many times, whatever version, keeps giving that same error :|.

Won't try fixes anymore as I can't afford to reinstall my system once again.

**Comment by fjperini** [comment]
@BorisDess use `brew install osgeo-qgis` (with R, SAGA, GRASS, ORFEO, others)

**Comment by eldang** [comment]
I gave up entirely and went back to Kyngchaos's installers for QGIS.  I have tried every workaround suggested in this thread, and a complete uninstall-and-reinstall of homebrew, plus fixing Python at 3.6.x.  When I was able to get QGIS working, the next update would break it again.

I'm sorry I don't have anything more constructive to add.  I really appreciate the work you're doing with this.  But in it's current form it's not usable.

**Comment by jonathanmccormack** [comment]
For what it's worth, I ran into similar issues a while back, and after banging my head against a wall for quite some time I realized I had run `brew switch` to use an older version of qt5-webkit at some point, which was breaking things.  Because I had the correct version of qt5-webkit installed (it just was not correctly linked) I didn't notice the problem until I finally uninstalled and reinstalled everything.

**Comment by fjperini** [comment]
@eldang @jonathanmccormack Keep in mind that these types of problems are due to your homebrew installation, which is external to this tap.

QGIS is working correctly. Tested on many occasions and by several users.

For a better analysis send the logs.

You can check it, a view of the last release:


![qgis](https://user-images.githubusercontent.com/21315242/58736062-e29fb900-83d2-11e9-802a-d12ae77b9adc.gif)

**Comment by BorisDess** [comment]
Hello.
I'm using a windows machine.
I don't think I installed homebrew...
Should I install then Git for Windows?

**Comment by eldang** [comment]
@BorisDess If you're not using Homebrew then this repository isn't the relevant place for you to get help or report bugs - it's specifically a project about porting a set of OS GIS tools to Homebrew.  The QGIS project has its own Github organization at https://github.com/qgis

**Comment by BorisDess** [comment]
Thanks, you are right.
I'm sorry, the only reason being, I have the same exact error than the user. 
My bad.

**Comment by fjperini** [comment]
@BorisDess homebrew is a package manager for macOS. in the git repository of QGIS you can find support for the problems that you are having in windows.

@eldang @dd-a @jonathanmccormack If you need more support, you can contact me through the space we have in slack.

**Comment by dd-a** [comment]
Thank you @fjperini , everything is fine for me since the last big reorganization you did recently.  All the problems were solved.  Thank you again very much !

**Comment by jonathanmccormack** [comment]
No issues here either -- as I said, I was able to resolve the problem by running `brew switch` to the correct, latest version of the dependency.  Everything is working great now.  Thanks for the hard work @fjperini!

**Comment by SLSJL** [comment]
In my case, it turns out to be a conflict of different versions of python.  I uninstalled python 3.7 after I installed the latest python 3.8. 
Somehow QGIS was still using python 3.7. And the folder `python 3.7` reminds in the directory `C:\Users\user\AppData\Roaming\Python`.

Problem solved by deleting the python 3.7 folder.

BTW, it is a case on Windows
